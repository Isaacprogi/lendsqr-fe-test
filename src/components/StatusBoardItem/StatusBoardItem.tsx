import { statusBoardItemProps } from '../../utils/types/interface_and_types'

export const StatusBoardItem = ({logo,title,value}:statusBoardItemProps) => {
  return (  

    <div >
    <span className="circle">
        <img src={logo} alt="status-board-logo" />
    </span>
    <span className="title">{title}</span>
    <span className="value">{value}</span>
</div>
  )
}
