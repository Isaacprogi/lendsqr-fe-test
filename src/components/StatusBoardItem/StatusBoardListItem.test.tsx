import { fireEvent, render, screen } from '@testing-library/react';
import {StatusBoardItem} from './StatusBoardItem'
import Img from '../../assets/images/isaac.jpg'
import renderer from 'react-test-renderer'

describe('test for NavBar component', ()=>{
   
    test('props shoud be present', ()=>{
        render(<StatusBoardItem logo={Img} title='Users' value='2000'  />)
        expect(screen.getByRole('img',{name:'status-board-logo'})).toHaveAttribute('src', Img)
        expect(screen.getByText(/Users/i)).toBeInTheDocument()
        expect(screen.getByText(/2000/i)).toBeInTheDocument()
    })  

    test('should match snapshot', async () => {
        const tree = renderer.create(<StatusBoardItem logo={Img} title='Users' value='2000'  />).toJSON();
        expect(tree).toMatchSnapshot();
      });
})