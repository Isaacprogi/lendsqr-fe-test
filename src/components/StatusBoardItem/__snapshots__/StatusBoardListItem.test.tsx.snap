// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test for NavBar component > should match snapshot 1`] = `
<div>
  <span
    className="circle"
  >
    <img
      alt="status-board-logo"
      src="/src/assets/images/isaac.jpg"
    />
  </span>
  <span
    className="title"
  >
    Users
  </span>
  <span
    className="value"
  >
    2000
  </span>
</div>
`;

exports[`test for NavBar component should match snapshot 1`] = `
<div>
  <span
    className="circle"
  >
    <img
      alt="status-board-logo"
      src="isaac.jpg"
    />
  </span>
  <span
    className="title"
  >
    Users
  </span>
  <span
    className="value"
  >
    2000
  </span>
</div>
`;
